source/agora/api/FullNode.d

```diff
@@ -201,23 +204,4 @@ public interface API
 
     @method(HTTPMethod.GET)
     public bool hasEnrollment (Hash enroll_hash);
-
-    /***************************************************************************
-
-        Get validator's pre-image inforamtion
-
-        API:
-            GET /get_preimage
-
-        Params:
-            enroll_key = The key for the enrollment in which the pre-image is
-                contained.
-
-        Returns:
-            preimage information of the validator if exists, otherwise
-                PreImageInfo.init
-
-    ***************************************************************************/
-
-    public PreImageInfo getPreimage (Hash enroll_key);
```

Comment:
This doesn't look right either. The full nodes should be able to propagate preimages too.

Reply:
They're back in place.

Reply:
This PR now spreads preimages, including full nodes.